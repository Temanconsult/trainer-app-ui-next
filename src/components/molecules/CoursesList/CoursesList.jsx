import React from 'react'
import PropTypes from 'prop-types'

import { Row } from 'react-bootstrap'
import CourseInfoCard from '../CourseInfoCard'

const CoursesList = ({ courses, className, showHeading }) => {
  return (
    <Row className={`${className} mt-4 mx-auto`}>
      {showHeading && <h2>Courses</h2>}
      <Row className="courses">
        {courses.map((course) => {
          return <CourseInfoCard key={course.name} course={course} />
        })}
      </Row>
    </Row>
  )
}

CoursesList.propTypes = {
  courses: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
    })
  ),
  className: PropTypes.string,
  showHeading: PropTypes.bool,
}

CoursesList.defaultProps = {
  courses: [],
  className: '',
  showHeading: false,
}

export default CoursesList
