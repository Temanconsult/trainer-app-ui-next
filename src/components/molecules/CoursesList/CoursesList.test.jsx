import React from 'react'
import { render } from '@testing-library/react'
import CoursesList from '.'

describe('CoursesList', () => {
  it('creates courses list', () => {
    const { asFragment } = render(
      <CoursesList
        courses={[
          { name: 'Name 1', description: 'Description 1' },
          { name: 'Name 2', description: 'Description 2' },
        ]}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
