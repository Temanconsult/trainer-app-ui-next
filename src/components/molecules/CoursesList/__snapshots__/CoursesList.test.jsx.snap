// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoursesList creates courses list 1`] = `
<DocumentFragment>
  <div
    class=" mt-4 mx-auto row"
  >
    <div
      class="courses row"
    >
      <div
        class="mb-4 mr-md-2 mr-lg-4 mr-sm-0 card bg-light text-dark"
        style="max-width: 30rem;"
      >
        <div
          class="card-body"
        >
          <img
            class="card-img-top"
            src="file mock"
          />
          <div
            class="card-title h5"
          >
            Name 1
          </div>
          <p
            class="card-text"
          >
            Description 1
          </p>
        </div>
      </div>
      <div
        class="mb-4 mr-md-2 mr-lg-4 mr-sm-0 card bg-light text-dark"
        style="max-width: 30rem;"
      >
        <div
          class="card-body"
        >
          <img
            class="card-img-top"
            src="file mock"
          />
          <div
            class="card-title h5"
          >
            Name 2
          </div>
          <p
            class="card-text"
          >
            Description 2
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
